frontend/src/components/Searchbar/useSearchbar.ts

```diff
@@ -0,0 +1,51 @@
+/**
+ * Hook exposing convenient variables for use with `Searchbar` component.
+ */
+
+import {
+  RefObject,
+  useCallback,
+  useLayoutEffect,
+  useRef,
+  useState,
+} from 'react'
+
+type UseSearchbarReturn = {
+  inputRef: RefObject<HTMLInputElement>
+  isExpanded: boolean
+  handleExpansion: () => void
+  handleCollapse: () => void
+}
+
+export const useSearchbar = ({
+  isInitiallyExpanded = false,
+  isFocusOnExpand = false,
+}: {
+  /**
+   * If `true`, the searchbar will be expanded on initial render.
+   */
+  isInitiallyExpanded?: boolean
+  /**
+   * If `true`, the searchbar will be focused whenever the searchbar is expanded.
+   */
+  isFocusOnExpand?: boolean
+}): UseSearchbarReturn => {
+  const [isExpanded, onExpansionChange] = useState(isInitiallyExpanded)
+  const inputRef = useRef<HTMLInputElement>(null)
+
+  useLayoutEffect(() => {
```

Comment:
do we really need `useLayoutEffect` here? the effect seems to depend on the component props and state, so it doesn't seem to need the DOM to be mutated before running. tbh unnecessarily using `useLayoutEffect` instead of `useEffect` smells like an antipattern since it blocks visual updates